src/nxdoc/nuxeo-server/administration/variables-and-configuration/variables-and-configuration-pattern.md

```diff
@@ -0,0 +1,300 @@
+This environment is retrieved by `nuxeo-launcher` and by `nuxeo-connect-standalone`, and is useful for Nuxeo Packages management logics.
+
+### DeploymentPreprocessor Templates Variables Expansion
+
+Java class [DeploymentPreprocessor](https://github.com/nuxeo/nuxeo/blob/v11.4.31/modules/runtime/nuxeo-runtime-deploy/src/main/java/org/nuxeo/runtime/deployment/preprocessor/DeploymentPreprocessor.java) logics are also covered by variable expansion using the following patterns:
+
+``` 
+${propertyName}
+```
+
+Or:
+
+``` 
+${propertyName:=defaultValue}
+``` 
+
+The map used for expansion is a bit different and includes the following:
+1. A [default environment](https://github.com/nuxeo/nuxeo/blob/v11.4.31/modules/runtime/nuxeo-runtime-osgi/src/main/java/org/nuxeo/osgi/application/loader/FrameworkLoader.java#L141) is created, using a map filled with System properties, and some selected additional properties are [added to this map](https://github.com/nuxeo/nuxeo/blob/v11.4.31/modules/runtime/nuxeo-runtime-osgi/src/main/java/org/nuxeo/osgi/application/loader/FrameworkLoader.java#L316).
+1. Then the map is filled with properties defined in the `NUXEO_HOME/nxserver/system.properties` files.
+1. Then the properties computed during pre-processing are added.
+
+These steps are covered by environment variables expansion explained below.
+
+Variable expansion happens when running copy/append commands (and alike) declared in the deployment-fragment.xml file: content is processed to replace variables using the resulting properties map. The replacement is recursive in case the property references another property.
+
+This is useful to control configuration added to the `web.xml`, or other preprocessor template files defined in `NUXEO_HOME/nxserver/META-INF` like `faces-config.xml`. Here’s an example from JSF:
+
+```xml
+<context-param>
+  <param-name>javax.faces.FACELETS_REFRESH_PERIOD</param-name>
+  <param-value>${javax.faces.FACELETS_REFRESH_PERIOD:=-1}</param-value>
+</context-param>
+``` 
+
+{{#> callout type='note' heading='DeploymentPreprocessor logics'}}
+
+`DeploymentPreprocessor` logics are even harder to decipher than others, so not sure that the files copied to the `nuxeo.war` go through variable expansion logics, for instance (useful for i18n messages or static CSS files management, typically).
+
+{{/callout}}
+
+
+## Nuxeo Runtime Startup
+
+The properties map available to the runtime framework after startup differs from the one handled during pre-processing: let’s start by explaining how this property map is defined, and what kind of variable expansion is supported.
+
+Then let’s summarize the various options to define and retrieve properties for the Nuxeo Runtime XML Components configurations (that have been documented [here]({{page version='' space='nxdoc' page='how-to-create-a-service/#defining-properties-on-the-service-component'}}).
```

Comment:
broken link with anchor (and missing closing ")"), i believe it should be:

`(that have been documented [here]({{page page='how-to-create-a-service'}}#defining-properties-on-the-service-component))`